daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dpath/NodeInfo.scala

```diff
@@ -993,6 +1001,7 @@ object NodeInfo extends Enum {
   lazy val allTypes =
     allDFDLTypes ++ List(
       Complex,
+      Scalar,
       Array,
```

Comment:
I notice Exists is listed in allAbstractTypes (above at line 969) while Scalar is listed in allTypes here.  I'm sure Scalar is more analogous to Complex than Exists, but I thought I should mention it.

Reply:
Good point. Scalar's children are Complex (here in `allTypes`) and AnySimple (above in allAbstractTypes), so there doesn't seem to be an obvious place for. I don't think it currently matters because nothing actually uses `allAbstractTypes`, but it feels like Scalar is more like Exists/AnySimpleType than Complex and does want to be in the abstract lis.

Reply:
Moved to the abstract list